src/vs/workbench/contrib/externalTerminal/browser/externalTerminal.contribution.ts

```diff
@@ -26,114 +26,195 @@ import { LifecyclePhase } from 'vs/workbench/services/lifecycle/common/lifecycle
 import { Registry } from 'vs/platform/registry/common/platform';
 import { IExternalTerminalConfiguration, IExternalTerminalService } from 'vs/platform/externalTerminal/common/externalTerminal';
 import { TerminalLocation } from 'vs/platform/terminal/common/terminal';
+import { IDisposable } from 'xterm';
 
-const OPEN_IN_TERMINAL_COMMAND_ID = 'openInTerminal';
-CommandsRegistry.registerCommand({
-	id: OPEN_IN_TERMINAL_COMMAND_ID,
-	handler: async (accessor, resource: URI) => {
-		const configurationService = accessor.get(IConfigurationService);
-		const editorService = accessor.get(IEditorService);
-		const fileService = accessor.get(IFileService);
-		const integratedTerminalService = accessor.get(IIntegratedTerminalService);
-		const remoteAgentService = accessor.get(IRemoteAgentService);
-		const terminalGroupService = accessor.get(ITerminalGroupService);
-		let externalTerminalService: IExternalTerminalService | undefined = undefined;
-		try {
-			externalTerminalService = accessor.get(IExternalTerminalService);
-		} catch {
-		}
+const OPEN_IN_INTEGRATED_TERMINAL_COMMAND_ID = 'openInIntegratedTerminal';
+const OPEN_IN_EXTERNAL_TERMINAL_COMMAND_ID = 'openInExternalTerminal';
```

Comment:
We definitely need to keep the old command in order to not break custom keybindings. How about we have 3 commands all with different `when` clauses:

- openInTerminal (shows when integrated or external)
- openInIntegratedTerminal (shows when both)
- openInExternalTerminal (shows when both)

Reply:
Let me give it a try

Reply:
@Tyriar I made some modifications, I found that if only two items are maintained, it will reduce the complexity and code size a little, the disadvantage is that you need to add a state representing the history of the command to keep the history of the command available. I haven't optimized the update logic for the title, currently they only show up as 'Open in Integrated Terminal' and 'Open in External Terminal', I need some advice from you, do you think we need to keep three items?

Reply:
I guess it could work with 2 commands, `openInTerminal` and `openInIntegratedTerminal`, the first one will always open the external terminal unless it's not possible (eg. remote workspace).